packages/client/src/createTRPCClient.ts

```diff
@@ -115,37 +123,29 @@ export const clientCallTypeToProcedureType = (
   return clientCallTypeMap[clientCallType as keyof typeof clientCallTypeMap];
 };
 
-/**
- * Creates a proxy client and shows type errors if you have query names that collide with built-in properties
- */
-export type CreateTRPCClient<TRouter extends AnyRouter> =
-  inferRouterClient<TRouter> extends infer $Value
-    ? UntypedClientProperties & keyof $Value extends never
-      ? inferRouterClient<TRouter>
-      : IntersectionError<UntypedClientProperties & keyof $Value>
-    : never;
-
 /**
  * @internal
  */
 export function createTRPCClientProxy<TRouter extends AnyRouter>(
   client: TRPCUntypedClient<TRouter>,
-): CreateTRPCClient<TRouter> {
-  const proxy = createRecursiveProxy<CreateTRPCClient<TRouter>>(
+): inferRouterClient<TRouter> {
+  const proxy = createRecursiveProxy<inferRouterClient<TRouter>>(
     ({ path, args }) => {
       const pathCopy = [...path];
       const procedureType = clientCallTypeToProcedureType(pathCopy.pop()!);
 
       const fullPath = pathCopy.join('.');
 
-      return (client as any)[procedureType](fullPath, ...args);
+      return (client[procedureType] as any)(fullPath, ...(args as any));
     },
   );
-  return createFlatProxy<CreateTRPCClient<TRouter>>((key) => {
-    if (client.hasOwnProperty(key)) {
-      return (client as any)[key as any];
-    }
-    if (key === '__untypedClient') {
+  return createFlatProxy<inferRouterClient<TRouter>>((key) => {
+    if (
+      key === untypedClientSymbol ||
+      // Safer for monorepos?:
+      (typeof key === 'symbol' &&
+        key.description === untypedClientSymbol.description)
```

Comment:
I may be a bit overly defensive, but I don't trust symbols in monorepos (unless they're pnpm)

Maybe this isn't needed

Reply:
May be easier just to use `Symbol.for("trpcUntypedClient")` and rely on the global symbol registry, rather than comparing descriptions like this?

Reply:
hmmm https://tsplay.dev/NDpq1N

how do you mean?

![CleanShot 2025-02-13 at 01 17 42@2x](https://github.com/user-attachments/assets/e4619ee5-fff8-4a0b-85cb-d65d18a2289b)

Reply:
Sorry! Here's the context I should have provided first time around:

```ts
const untypedClientSymbol = Symbol.for("trpcUntypedClient");
```

Then you have a guarantee that `key === untypedClientSymbol` will always be true, even if there's a monorepo that has somehow bundled two copies of TRPC (symbol lives in a shared registry, instead of in a module).

Reply:
Ohhh, I didn't know that trick, awesome!